mithril-signer/src/database/migration.rs

```diff
@@ -93,6 +93,30 @@ insert into stake_pool (epoch, stake_pool_id, stake, created_at)
     from stake, json_each(stake.value) as stake_dis 
     order by epoch asc;
 drop table stake;
+"#,
+        ),
+        // Migration 5
+        // Add the `protocol_initializer` table and migration data from the previous
+        // `protocol_initializer` JSON format.
+        SqlMigration::new(
+            5,
+            r#"
+create table new_protocol_initializer (
+    epoch         integer   not null,
+    protocol      json      not null,
+    created_at    text      not null,
+    primary key (epoch)
+);
+create table if not exists protocol_initializer (key_hash text primary key, key json not null, value json not null);
```

Comment:
Why do we need to recreate the legacy table here? Is it to allow running the migration step individually in tests?

Reply:
This is the pattern used for `stake_pool` migration from `stake` in `aggregator`.
This makes sure that the table exists for the migration query that follows.
I suppose that is because this kind of  table are created by the `StoreAdapter` code when it's used and may have not be created during tests that don't use it.
If we remove this line, the script will crash if we try to execute it in an empty database.